pkg/minikube/constants/constants.go

```diff
@@ -215,33 +218,68 @@ var LocalkubeCachedImages = []string{
 	"gcr.io/k8s-minikube/storage-provisioner:v1.8.0",
 }
 
-func GetKubeadmCachedImages(version string) []string {
-	return []string{
-		// Dashboard
-		"k8s.gcr.io/kubernetes-dashboard-amd64:v1.8.1",
-
-		// Addon Manager
-		"k8s.gcr.io/kube-addon-manager:v8.6",
+func GetKubeadmCachedImages(kubernetesVersionStr string) []string {
 
-		// Pause
-		"k8s.gcr.io/pause-amd64:3.1",
+	var images = []string{
+		"k8s.gcr.io/kube-proxy-amd64:" + kubernetesVersionStr,
+		"k8s.gcr.io/kube-scheduler-amd64:" + kubernetesVersionStr,
+		"k8s.gcr.io/kube-controller-manager-amd64:" + kubernetesVersionStr,
+		"k8s.gcr.io/kube-apiserver-amd64:" + kubernetesVersionStr,
+	}
 
-		// DNS
-		"k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8",
-		"k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.8",
-		"k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.8",
+	v1_10 := semver.MustParseRange(">=1.10.0 <1.11.0")
+	v1_9 := semver.MustParseRange(">=1.9.0 <1.10.0")
+	v1_8 := semver.MustParseRange(">=1.8.0 <1.9.0")
 
-		// etcd
-		"k8s.gcr.io/etcd-amd64:3.1.12",
+	kubernetesVersion, err := semver.Make(strings.TrimPrefix(kubernetesVersionStr, minikubeVersion.VersionPrefix))
+	if err != nil {
+		glog.Errorln("Error parsing version semver: ", err)
+	}
 
-		"k8s.gcr.io/kube-proxy-amd64:" + version,
-		"k8s.gcr.io/kube-scheduler-amd64:" + version,
-		"k8s.gcr.io/kube-controller-manager-amd64:" + version,
-		"k8s.gcr.io/kube-apiserver-amd64:" + version,
+	if v1_10(kubernetesVersion) {
```

Comment:
One last nit, it might be a bit more clear if this first if statement is something like:

`if v_10 || greater_than_v_10`

that makes it clearer that versions outside the range (but higher) get the same images as the latest.

Reply:
Sure. Updated.